test/coverage-test/test/coverage.test.ts

```diff
@@ -1,8 +1,11 @@
 import { expect, test } from 'vitest'
-import { pythagoras } from '../src/index.mjs'
 import { implicitElse } from '../src/implicitElse'
 import { useImportEnv } from '../src/importEnv'
 
+// TODO Fix: Browser fails to load if extensions ".mjs" is used
+// @ts-expect-error -- extension
+import { pythagoras } from '../src/index'
+
```

Comment:
I'll look into this next. Any pointers are appreciated. 🙌  
Do we set any specific settings on Node side for this that is not included when running in browser?


```
// Filename on filesystem is "../src/index.mts"
import { pythagoras } from '../src/index.mjs'
```

```
12:34:25 PM [vite] Internal server error: Failed to resolve import "../src/index.mjs" from "test/coverage.test.ts?v=1679654065071". Does the file exist?
  Plugin: vite:import-analysis
  File: /x/vitest/test/coverage-test/test/coverage.test.ts?v=1679654065071:7:27
  2  |  import { implicitElse } from "../src/implicitElse";
  3  |  import { useImportEnv } from "../src/importEnv";
  4  |  import { pythagoras } from "../src/index.mjs";
     |                              ^
  5  |  test("Math.sqrt()", async () => {
  6  |    expect(pythagoras(3, 4)).toBe(5);
      at formatError (file:///x/vitest/node_modules/.pnpm/vite@4.0.0_@types+node@18.7.13/node_modules/vite/dist/node/chunks/dep-ed9cb113.js:42301:46)
      at TransformContext.error (file:///x/vitest/node_modules/.pnpm/vite@4.0.0_@types+node@18.7.13/node_modules/vite/dist/node/chunks/dep-ed9cb113.js:42297:19)
      at normalizeUrl (file:///x/vitest/node_modules/.pnpm/vite@4.0.0_@types+node@18.7.13/node_modules/vite/dist/node/chunks/dep-ed9cb113.js:40600:33)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async TransformContext.transform (file:///x/vitest/node_modules/.pnpm/vite@4.0.0_@types+node@18.7.13/node_modules/vite/dist/node/chunks/dep-ed9cb113.js:40753:47)
      at async Object.transform (file:///x/vitest/node_modules/.pnpm/vite@4.0.0_@types+node@18.7.13/node_modules/vite/dist/node/chunks/dep-ed9cb113.js:42572:30)
      at async loadAndTransform (file:///x/vitest/node_modules/.pnpm/vite@4.0.0_@types+node@18.7.13/node_modules/vite/dist/node/chunks/dep-ed9cb113.js:40378:29)
```

Reply:
> Do we set any specific settings on Node side for this that is not included when running in browser?

No, it should work out of the box as far as I'm concerned. You can try doing a reproduction in https://vite.new with a fresh VIte project

Reply:
I don't think they are necessary anymore, since we reload the page on watch change. And these were there to allow revalidation.

Reply:
After some testing it seems that we need these `v` query parameters. In watch mode the changes are not reloaded on re-runs. I guess even the browser cache could be serving the old file.

We can leave this TODO here as is and come back to it later. After looking at this more think this is a bug in `vite:resolve` plugin - I'll keep debugging that one and see if fix could be applied later.

Reply:
> We can leave this TODO here as is and come back to it later

We can also just strip `v` in `resolveId` hook before `vite:resolve` kicks in.